src/main/java/org/example/StudentRepository.java

```diff
@@ -0,0 +1,122 @@
+package org.example;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class StudentRepository {
+
+    List<Student> studentList = new ArrayList<>();
+    LocalDate curentDate = LocalDate.now();
+    private int studentAge = 0;
+
+
+    /**
+     * method for adding a student in our list with validation for name, date of birth, gender and id
+     */
+
+    public List<Student> addStudent(String firstName, String lastName, LocalDate birthDate, String genderType, String id) throws ValidationException {
+
+        if (birthDate.getYear() < 1900 || birthDate.getYear() > curentDate.getYear() - 18) {
```

Comment:
you can gather all the validation in to a validateStudent(student) method which you can surround with try/catch